src/artifacts.ts

```diff
@@ -41,7 +41,7 @@ export class Template extends Class {
             aggregate: specs?.aggregate || Mode(),
             tally: specs?.tally || Mode(),
         })
-        this.argsCount = specs.retrieve.map(retrieval => retrieval?.argsCount).reduce((prev, curr) => prev + curr) || 0
+        this.argsCount = specs.retrieve.map(retrieval => retrieval?.argsCount).reduce((prev, curr) => Math.max(prev, curr), 0) || 0
```

Comment:
You can directly call: `Math.max(...your_array)`. If it's a bit difficult to read in a single line, consider move `specs.retrieve.map(retrieval => retrieval?.argsCount)` to a variable.

```suggestion
        this.argsCount = Math.max([...specs.retrieve.map(retrieval => retrieval?.argsCount)]) || 0
```

Reply:
proposed line is more generic against possibility of specs.retrieve containing empty array